Cargo.toml

```diff
@@ -32,3 +32,8 @@ incremental = true
 # The profile that 'cargo dist' will build with
 [profile.dist]
 inherits = "release"
+
+[workspace.dependencies]
+log = "0.4"
+flexi_logger = "0.28"
```

Comment:
Would it be a huge burden to split the logging changes into a separate PR? It can be a bit difficult to review when there are multiple, unconnected changes in the same PR.

Reply:
While developing these new use cases for `target_os` I needed to observe how I was evaluating each in detail and that required debug logging. I was putting `println` but these debug log statements were something I wanted to retain in case I need to look into what is going wrong in the future. I decided we should have proper logging so debug logging can be retained and enabled. So the reason I added the log dependency is for this PR. I would not want to separate this into another PR.

Reply:
That's fair, and in this case I think it's small enough that it's not too much context switching when reviewing. In general though, I've found that getting timely reviews is easier when the changes are as small as possible. For example, this could be broken up by first having a PR that adds `log` and `flexi-logger` and converts all of the existing `println!` / `eprintln!` calls, then a separate PR stacked on top of that which adds the new behavior along side new uses of `debug!`.